server/src/main/java/org/opensearch/action/search/SearchResponse.java

```diff
@@ -417,6 +427,35 @@ public static SearchResponse innerFromXContent(XContentParser parser) throws IOE
                         }
                     }
                     clusters = new Clusters(total, successful, skipped);
+                } else if (EXT_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    extBuilders = new ArrayList<>();
+                    String extSectionName = null;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                        if (token == XContentParser.Token.FIELD_NAME) {
+                            extSectionName = parser.currentName();
+                        } else {
+                            SearchExtBuilder searchExtBuilder;
+                            try {
+                                searchExtBuilder = parser.namedObject(SearchExtBuilder.class, extSectionName, null);
+                                if (!searchExtBuilder.getWriteableName().equals(extSectionName)) {
+                                    throw new IllegalStateException(
+                                        "The parsed ["
+                                            + searchExtBuilder.getClass().getName()
+                                            + "] object has a "
+                                            + "different writeable name compared to the name of the section that it was parsed from: found ["
+                                            + searchExtBuilder.getWriteableName()
+                                            + "] expected ["
+                                            + extSectionName
+                                            + "]"
+                                    );
+                                }
+                                extBuilders.add(searchExtBuilder);
+                            } catch (XContentParseException e) {
+                                log.warn("Unable to detect SearchExtBuilder for {}: {}", extSectionName, e);
+                                parser.skipChildren();
```

Comment:
Ok, so this is an issue, @austintlee @msfroh I would suggested to not skip it (and also not just log it), but return the parsing result back to the client indicating the error, something  along this lines:

```
"ext": {
  "ext_1": {
     "failure": <exception here>
  }
}
```

Reply:
But this would only happen in an environment where your application does not have a particular SearchExtBuilder in its classpath or you are running an older version of a plug that doesn't have it.

If we add a failure reason, that is yet another state we have to maintain in SearchResponse and every ext builder will need to know how to write it out.

Reply:
This is correct, we could have only 1 SearchExtBuilder builder that handles failure reason - it does not depend on `ext` it reports an issue against. If we don't expose the problem, it will go unnoticed.

Reply:
But isn't that catch-22?  You want another ext builder to handle a failure for a missing ext builder?  Are you suggesting we add a SearchExtBuilder in the core to handle this case?  In the case of SearchRequest, this exception will just get propagated out.  Wouldn't that be the simplest way to handle this? @msfroh what do you think?

Reply:
>  Are you suggesting we add a SearchExtBuilder in the core to handle this case? In

Correct

>  In the case of SearchRequest, this exception will just get propagated out. Wouldn't that be the simplest way to handle this?

This is ideal but it this is breaking change - we can do that in 3.x but not 2.x

Reply:
@austintlee think a bit more about it in context of 

>  In the case of SearchRequest, this exception will just get propagated out. 

The `ext` in response is mirroring the `ext` in request. It is possible that use sends nothing in request `ext` section but could get back non-empty `ext` section in response?